### ISSUE TITLE
Does zazu have a constant internet connection?

### ISSUE THREAD

**Issue by tdoerrich** [tinytacoteam/zazu#350]
I guess this is more a question than a bug-report. Is zazu always connected to internet? Since each time I reload the config it "loads" the plugins. Are they all part of the app installer, or is it constantly downloading them and checking for updates?

**Comment by OscarBarrett** [comment]
Your plugins are installed locally. When the plugins are loaded by zazu (such as when reloading the config), zazu runs a [clone](https://github.com/tinytacoteam/zazu/blob/2da7485efa8706da0a730533394b4f268788fe30/app/lib/git.js#L45) function for each plugin but that won't actually do anything if the plugin is already cloned.

On a completely fresh install, you don't have any plugins and they are installed as part of the load process.

You should be able to see all your installed plugins in ~/.zazu/plugins. AFAIK plugins are only updated if you manually click on "Update Plugins" in the menu.

**Comment by linonetwo** [comment]
Yeah, as @OscarBarrett  explained, it won't download plugin code from Github again, nor install npm packages again.

But it will load plugins one by one https://github.com/tinytacoteam/zazu/blob/e6e37e63841ca4e8abd6bd6dfc7d211ae30d18d5/app/containers/pluginWrapper.js#L100

And each plugin load blocks one by one https://github.com/tinytacoteam/zazu/blob/e6e37e63841ca4e8abd6bd6dfc7d211ae30d18d5/app/packages/plugin.js#L79

And each block load it's JS code from FS https://github.com/tinytacoteam/zazu/blob/e6e37e63841ca4e8abd6bd6dfc7d211ae30d18d5/app/blocks/input/prefixScript.js#L14

So I guess the reason why it is slow and made you think it is download something, is because it is doing too much sync IO to disk.

There can be optimization on this, but not very necessary now.

Hope this solves your confusion.
